panic on invalid net.beta.kubernetes.io/network-policy annotation

I set the `net.beta.kubernetes.io/network-policy` annotation to an invalid (json) value:
```
kubectl annotate ns test "net.beta.kubernetes.io/network-policy={\"ingress\": {\"isolation\": \"DefaultDeny\"}}"- --overwrite
```

This caused kube-router to panic and crashloop, but only on the nodes that had pods in the namespace the policy was applied to.  Fixing the json in the annotation cleared things up.

I think I've found an unhandled Error that will lead me to a fix.  I'll submit a PR soon if I find it.

Log:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4eec82]

goroutine 205 [running]:
panic(0x1645840, 0xc420018050)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cloudnativelabs/kube-router/app/controllers.(*NetworkPolicyController).syncPodFirewallChains(0xc42027a100, 0xc42089e7e0, 0x0, 0x0)
        /home/kube/go/src/github.com/cloudnativelabs/kube-router/app/controllers/network_policy_controller.go:323 +0x112
github.com/cloudnativelabs/kube-router/app/controllers.(*NetworkPolicyController).Sync(0xc42027a100)
        /home/kube/go/src/github.com/cloudnativelabs/kube-router/app/controllers/network_policy_controller.go:160 +0x2ac
github.com/cloudnativelabs/kube-router/app/controllers.(*NetworkPolicyController).OnPodUpdate(0xc42027a100, 0xc4203981f0)
        /home/kube/go/src/github.com/cloudnativelabs/kube-router/app/controllers/network_policy_controller.go:112 +0x17a
github.com/cloudnativelabs/kube-router/app/watchers.(*podWatcher).RegisterHandler.func1(0x1540e00, 0xc4203981f0)
        /home/kube/go/src/github.com/cloudnativelabs/kube-router/app/watchers/pods_watcher.go:65 +0x4a
github.com/cloudnativelabs/kube-router/utils.ListenerFunc.OnUpdate(0xc42066b800, 0x1540e00, 0xc4203981f0)
        /home/kube/go/src/github.com/cloudnativelabs/kube-router/utils/utils.go:14 +0x3a
created by github.com/cloudnativelabs/kube-router/utils.(*Broadcaster).Notify
        /home/kube/go/src/github.com/cloudnativelabs/kube-router/utils/utils.go:37 +0xa3
```

--- a/app/controllers/network_policy_controller.go
+++ b/app/controllers/network_policy_controller.go
@@ -320,6 +320,9 @@ func (npc *NetworkPolicyController) syncPodFirewallChains() (map[string]bool, er
 
 	// loop through the pods running on the node which has default ingress to be denied
 	podsOnNodeInfo, err := getPodsRunningOnNode(npc.nodeIP.String())
+	if err != nil {
+		return nil, err
+	}
 	for _, pod := range *podsOnNodeInfo {
 
 		// below condition occurs when we get trasient update while removing or adding pod
@@ -623,6 +626,9 @@ func getNameSpaceDefaultPolicy(namespace string) (string, error) {
 				err := json.Unmarshal([]byte(networkPolicyAnnotation), &annot)
 				if err == nil {
 					return annot["ingress"]["isolation"], nil
+				} else {
+					glog.Errorf("Skipping invalid network-policy for namespace \"%s\": %s", namespace, err)
+					return "DefaultAllow", errors.New("Invalid NetworkPolicy.")
 				}
 			} else {
 				return "DefaultAllow", nil

